.github/workflows/codspeed.yml

```
@@ -46,7 +46,11 @@ jobs:
       - name: cache rust
         uses: Swatinem/rust-cache@v2
 
-      - run: pip install . -v
+      - name: Install pydantic-core
+        # build-prod does not include MiMalloc because it bypasses Maturin
+        run: |
+          make build-prod
```

I thought about it more and realized I think the `make build-dev` thing doesn't work for me if it isn't already pip installed in editable mode. Maybe try:
```suggestion
          pip install -e .
          make build-prod
```

---

How will we know which version is installed then 🥲

---

Similarly, I guess we need to confirm that the MiMalloc version is what wheels are using

---

Not sure if it's worth it, but I would imagine you could add an attribute to the `pydantic_core._pydantic_core` module conditionally based on the mimalloc feature flag? (And then check for it when uploading the wheel if necessary?)

```diff
diff --git a/pydantic_core/_pydantic_core.pyi b/pydantic_core/_pydantic_core.pyi
index 6859a16..517d507 100644
--- a/pydantic_core/_pydantic_core.pyi
+++ b/pydantic_core/_pydantic_core.pyi
@@ -21,6 +21,7 @@ from _typeshed import SupportsAllComparisons
 
 __all__ = (
     '__version__',
+    'allocator',
     'build_profile',
     'SchemaValidator',
     'SchemaSerializer',
@@ -35,6 +36,7 @@ __all__ = (
     'list_all_errors',
 )
 __version__: str
+allocator: str
 build_profile: str
 
 T = TypeVar('T', default=Any, covariant=True)
@@ -167,7 +169,8 @@ class Url(SupportsAllComparisons):
     def query(self) -> 'str | None': ...
     @property
     def fragment(self) -> 'str | None': ...
-    def __init__(self, url: str) -> None: ...
+    @staticmethod
+    def __new__(cls, url: str) -> Url: ...
     def unicode_host(self) -> 'str | None': ...
     def query_params(self) -> 'list[tuple[str, str]]': ...
     def unicode_string(self) -> str: ...
diff --git a/src/lib.rs b/src/lib.rs
index 613220a..9e9229e 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -42,9 +42,20 @@ pub fn get_version() -> String {
     version.replace("-alpha", "a").replace("-beta", "b")
 }
 
+#[cfg(feature = "mimalloc")]
+fn get_allocator() -> String {
+    "mimalloc".to_owned()
+}
+
+#[cfg(not(feature = "mimalloc"))]
+fn get_allocator() -> String {
+    "default".to_owned()
+}
+
 #[pymodule]
 fn _pydantic_core(_py: Python, m: &PyModule) -> PyResult<()> {
     m.add("__version__", get_version())?;
+    m.add("allocator", get_allocator())?;
     m.add("build_profile", env!("PROFILE"))?;
     m.add_class::<PySome>()?;
     m.add_class::<SchemaValidator>()?;
```
(Haven't tested; also maybe it's overkill)

---

Looks like it’s using the default allocator for benchmarks now based just on the slowdowns 😂.

I was thinking something similar but it’s complex. How do you feel about moving forward without that and add ing it later if it’s really needed?